libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -68,6 +74,46 @@ proc waitRepliesOrTimeouts(
 
   return (receivedReplies, failedPeers)
 
+proc dispatchPutVal(
+    kad: KadDHT, peer: PeerId, key: Key, val: seq[byte]
+): Future[void] {.
+    async: (raises: [CancelledError, DialFailedError, ValueError, LPStreamError])
+.} =
+  info "dialing", me = kad.rtable.selfId, peer = peer, key = key, val = val[0 .. 3]
+  let conn = await kad.switch.dial(peer, KadCodec)
+
+  defer:
+    await conn.close()
+
+  let msg = Message(
+    msgType: MessageType.putValue,
+    record: some(Record(key: some(key.getBytes()), value: some(val))),
+  )
+
+  info "writing"
+  await conn.writeLp(msg.encode().buffer)
+
+  info "waiting reply"
+  let reply = Message.decode(await conn.readLp(MaxMsgSize)).tryGet()
+
+  if reply.msgType != MessageType.putValue:
+    raise newException(ValueError, "unexpected message type in reply: " & $reply)
+
+proc putVal*(
+    kad: KadDHT, val: seq[byte], replic: int
+): Future[seq[PeerId]] {.async: (raises: [CancelledError]).} =
+  let keyDat: array[IdLength, byte] = sha256.digest(val).data
+  let key = Key(kind: KeyType.Hashed, data: keyDat)
+
```

Comment:
In kademlia specs (the OG kademlia), something that is mentioned wrt "STORE" rpc method: "to ensure the persistence of the infor-
mation, nodes periodically republish les". "STORE" is the equivalent to "PUT_VALUE" in libp2p kademlia. We probably have to add this behavior too because due to network churn, nodes in which we published a value might end up dissapearing.